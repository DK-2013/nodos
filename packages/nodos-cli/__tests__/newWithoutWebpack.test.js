import fsp from 'fs/promises';
import path from 'path';
import os from 'os';
import { setTimeout } from 'timers/promises';
import runNew from '../lib/new/index.js';

let dir;

beforeEach(async () => {
  dir = await fsp.mkdtemp(path.join(os.tmpdir(), 'nodos-'));
});

test('nodos/new', async () => {
  const appPath = 'site';
  const projectRoot = path.join(dir, appPath);
  const options = { exitProcess: false, args: ['new', appPath, '--skip-install', '--skip-db', '--without', 'webpack'] };
  await runNew(dir, options);
  await setTimeout(1000);
  const configFiles = await fsp.readdir(path.join(projectRoot, '/config'));
  expect(configFiles).toMatchSnapshot();

  const packageJsonContent = await fsp.readFile(path.join(projectRoot, 'package.json'), 'utf-8');
  expect(packageJsonContent).toMatchSnapshot();
});
