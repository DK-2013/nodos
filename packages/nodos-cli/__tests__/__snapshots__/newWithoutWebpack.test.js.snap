// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodos/new 1`] = `
Array [
  "application.js",
  "environments",
  "routes.yml",
]
`;

exports[`nodos/new 2`] = `
"{
  \\"name\\": \\"site\\",
  \\"scripts\\": {
    \\"test\\": \\"jest\\",
    \\"start\\": \\"nodos server\\",
    \\"build\\": \\"webpack -p\\"
  },
  \\"jest\\": {
    \\"testEnvironment\\": \\"@nodosjs/jest-environment\\",
    \\"testPathIgnorePatterns\\": [
      \\"/config/\\",
      \\"/node_modules/\\"
    ]
  },
  \\"dependencies\\": {
    \\"@babel/core\\": \\"^7.14.5\\",
    \\"@babel/node\\": \\"^7.14.5\\",
    \\"@babel/preset-env\\": \\"^7.14.5\\",
    \\"@nodosjs/cli\\": \\"^0.0.57\\",
    \\"@nodosjs/core\\": \\"^0.0.57\\",
    \\"@nodosjs/db-prisma-extension\\": \\"^0.0.57\\",
    \\"@nodosjs/view-extension\\": \\"^0.0.57\\",
    \\"pg\\": \\"^8.5.3\\"
  },
  \\"devDependencies\\": {
    \\"@nodosjs/jest-environment\\": \\"^0.0.57\\",
    \\"babel-jest\\": \\"^27.0.6\\",
    \\"eslint\\": \\"^7.28.0\\",
    \\"eslint-plugin-babel\\": \\"^5.3.1\\",
    \\"eslint-plugin-jest\\": \\"^24.3.6\\",
    \\"jest\\": \\"^27.0.6\\",
    \\"jest-cli\\": \\"^27.0.6\\"
  }
}
"
`;
